### DISCUSSION TITLE
Flavor without gradle.properties

### DISCUSSION THREAD

**Discussion by AndrewLiis** [yshrsmz/BuildKonfig#174] [Q&A] [answered]
I use flavor to build with different configurations.  But I have a script that prepares a build for me, it does many different things, but it also builds the project with a specific flavor. After this build I run the project in Android Studio then the build is restarted and my BuildKonfig file is rewritten without flavor. The option using gradle.properties is not suitable for me because of the peculiarities of the script. Are there any other options how I can run the project through android studio without rewriting BuildKonfig object

**Comment by yshrsmz** [comment]
It's just a gradle property, you can set programatically with standard Gradle API.

**Reply by AndrewLiis** [reply]
> It's just a gradle property, you can set programatically with standard Gradle API.

The problem is that i have no way to change gradle.properties

**Reply by yshrsmz** [reply]
you can set a property from gradle file, without touching gradle.properties.

**Reply by yshrsmz** [reply, marked as answer]
@AndrewLiis Check this discussion if you want a quick code example
https://github.com/yshrsmz/BuildKonfig/discussions/128
